### ISSUE TITLE
App to app improvement

### ISSUE THREAD

**Issue by matjaz** [athombv/homey#659]
**What is your Homey version (Settings → System)?**
0.8.35

**What did you try to achieve?**
Request other app with specific version/functionality. Currently we need to send a request to another app and check response if functionality is available.

**What did you expect as result?**
Something like

``` js
var app = new api.App('com.youtube@^0.4.0')
app.version  '^0.4.0' // satisfies
```

If installed app does not satisfies version return `installed === false`.

**Any other remarks**
I bump into this use case connecting [YouTube](https://github.com/matjaz/homey-youtube) and [Chromecast](https://github.com/athombv/com.google.chromecast) apps.

Since Homey API is async I would prefer start using promises.
I would suggest alternative API for this:

``` js
api.getApp('com.youtube@^0.4.0')
  .then(app => {
    app === undefined // if not installed
    app.version // installed version
  })
  .catch(err => ..'something bad happened')
```

**Comment by nlrb** [comment]
Looks similar like #512.

**Comment by matjaz** [comment]
Yeah it's similar. But I like the 'satisfies' part so user doesn't need to explicitly check version.

**Comment by WeeJeWel** [comment]
Closing as this is fixed in v1.5.0
